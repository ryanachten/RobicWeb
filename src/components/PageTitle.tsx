import React from "react";
import classnames from "../utils";
import Typography from "@material-ui/core/Typography";
import { withStyles, Theme, createStyles } from "@material-ui/core";

type Props = {
  className?: string;
  classes: any;
  label: string;
};

const styles = (theme: Theme) =>
  createStyles({
    text: {
      margin: theme.spacing.unit * 4,
      marginLeft: 0,
      textTransform: "lowercase"
    }
  });

const PageTitle = ({ className, classes, label }: Props) => {
  return (
    <Typography
      className={classnames(classes.text, className)}
      variant="h5"
      component="h1"
    >
      {label}
    </Typography>
  );
};

export default withStyles(styles)(PageTitle);
