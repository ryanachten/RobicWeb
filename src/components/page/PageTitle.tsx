import React from "react";
import classnames from "../../utils";
import Typography from "@material-ui/core/Typography";
import { withStyles, Theme, createStyles } from "@material-ui/core";
import { Link } from "../Link";

type Props = {
  breadcrumb?: {
    label: string;
    onClick?: () => void;
    url?: string;
  };
  className?: string;
  classes: any;
  label: string;
};

const styles = (theme: Theme) =>
  createStyles({
    wrapper: {
      margin: theme.spacing(4),
      marginLeft: 0
    },
    text: {
      marginLeft: 0,
      textTransform: "lowercase"
    }
  });

const PageTitle = ({ className, classes, label, breadcrumb }: Props) => {
  return (
    <div className={classes.wrapper}>
      <Typography
        className={classnames(classes.text, className)}
        variant="h5"
        component="h1"
      >
        {label}
      </Typography>
      {breadcrumb && (
        <Link
          label={breadcrumb.label}
          url={breadcrumb.url}
          onClick={breadcrumb.onClick}
        />
      )}
    </div>
  );
};

const styled = withStyles(styles)(PageTitle);
export { styled as PageTitle };
