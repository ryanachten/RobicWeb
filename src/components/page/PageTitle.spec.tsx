import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import { PageTitle } from "./PageTitle";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Component: PageTitle", () => {
  let shallow: any;

  beforeEach(() => {
    shallow = createShallow({ untilSelector: "Select" });
  });

  const initProps = {
    breadcrumb: {
      label: "Testing",
      url: "/testing"
    },
    label: "Testing"
  };

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <PageTitle {...initProps} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
