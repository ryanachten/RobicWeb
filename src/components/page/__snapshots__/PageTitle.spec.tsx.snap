// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PageTitle renders correctly 1`] = `
<div
  className="PageTitle-wrapper-1"
>
  <h1
    className="MuiTypography-root-3 MuiTypography-h5-19 PageTitle-text-2"
  >
    Testing
  </h1>
  <a
    className="Link-link-39"
    href="/testing"
    onClick={[Function]}
  >
    <span
      className="MuiTypography-root-3 MuiTypography-body1-12 MuiTypography-colorInherit-32 MuiTypography-inline-38 MuiLink-root-40 MuiLink-underlineHover-42"
    >
      Testing
    </span>
  </a>
</div>
`;
